src/app.js

```diff
@@ -1,89 +1,110 @@
-        });
+  const getAddress = () => {
+    if(Object.keys(coords).length !== 0) {
+      const API_URL = `${process.env.REACT_APP_ADD_URL}q=${coords.latitude}+${coords.longitude}&key=${process.env.REACT_APP_ADD_KEY}`
+
+      fetch(API_URL)
+      .then(res => res.json())
+      .then(data => {
+        // console.log(data)
+        const addr = {}
+        if(data.results[0].components.city !== undefined)
+          addr['city'] = data.results[0].components.city
+        else {
+          addr['village'] = data.results[0].components.village
+        }
+        addr['countryCode'] = data.results[0].components['ISO_3166-1_alpha-2']
+        addr['country'] = data.results[0].components.country
+        // console.log(addr)
+        setAddress(addr)
       })
+      .catch(err => console.log(err))
+    }
   }
 
-  render() {
-    return <div className="weather-container"
-                style={{backgroundColor: this.state.color}}>
-      {_.isEmpty(this.state.weekWeather) ? "no data" :
-        <WeekWeather color={this.state.color}
-                     weekWeather={this.state.weekWeather} />}
-      {this._renderForm()}
-      <blockquote className="blockquote blockquote-centered">
-        <p>
-          Created by
-          <a href="http://twitter.com/mironcatalin"> @mironcatalin</a>
-        </p>
-      </blockquote>
-    </div>;
+  useEffect(getCoords, [])
+  useEffect(getAddress, [coords])
+
+  const getWeather = (searchedCity) => {
+    fetchWeather(searchedCity)
+    .then(res => {
+      const weatherData = res.list.map(weatherOfDay => weatherOfDay)
+      setWeather(weatherData)
+      setCity(searchedCity)
+      setCountry(res.city.country)
+      setColor(randomColor({luminosity: "dark", format: "hex"}))
+    })
   }
 
-  _renderForm() {
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    const searchedCity = event.target.value;
+
+    if (searchedCity === city) {
+      return;
+    }
+    setSearch(searchedCity)
+  }
+
+  const renderForm = () => {
     return (
-      <form onSubmit={(e) => this.handleSubmit(e)}>
+      <form onSubmit={handleSubmit}>
         <fieldset>
           <legend>Enter your city</legend>
-          <input className="form-input"
-                 ref="locationName"
-                 type="text"
-                 defaultValue={this.state.searchedCity}/>
+          <input className="form-input" ref="locationName" type="text" defaultValue={search}/>
         </fieldset>
       </form>
     );
   }
 
-  handleSubmit(e) {
-    e.preventDefault();
-
-    const searchedCity = this.refs.locationName.getDOMNode().value;
-
-    if (searchedCity === this.state.city) {
-      return;
-    }
-
-    this.getWeather(searchedCity);
-  }
-};
+  useEffect(getWeather, [search])
 
+  return (
+    <div className="weather-container" style={{backgroundColor: color}}>
+      {weather.length === 0 ? "no data" : <WeekWeather color={color} weather={weather} city={city} country={country}/>}
```

Comment:
Condition can be simply `{weather.length ? "no data" : <WeekWeather color={color} weather={weather} city={city} country={country}/>} `